monai/transforms/compose.py

```diff
@@ -106,8 +106,13 @@ def execute_compose(
         return data
 
     for _transform in transforms[start:end]:
-        if threading:
-            _transform = deepcopy(_transform) if isinstance(_transform, ThreadUnsafe) else _transform
+        with lock:
```

Comment:
We'll still need to create the lock object somewhere and get it to this function. Note that the lock must be created somewhere where only a single thread of execution is occurring